test/_RecorderTests.cpp

```diff
@@ -38,16 +38,17 @@ namespace RcclUnitTesting
 
     std::vector<rccl::rcclApiCall> calls;
     char entry[4096];
+    gethostname(entry, 256);
     //parse the outfile
-    std::string filename = "test" + std::to_string(pid) + ".json";
-    std::ifstream fp("/tmp/test" + std::to_string(pid) + ".json");
+    std::string filename = "/tmp/test." + std::to_string(pid) + "." + std::string(entry) + ".json";
+    std::ifstream fp(filename);
     fp.getline(entry, 4096);
     fp.getline(entry, 4096);
     fp.getline(entry, 4096);
     parseJsonEntry(entry, calls);
     int result = memcmp((char*)&calls[0]+4, (char*)&call+4, sizeof(rccl::rcclApiCall)-4);
```

Comment:
Seems you are trying to skip comparison of the `pid` field? Writing in this way is hard to understand and is brittle against changes to the struct. First suggestion, just copy the `pid` field from `calls[0]` into `call` and directly compare.
```suggestion
    call.pid = calls[0].pid; // pid will be different, do not compare original values
    int result = memcmp(&calls[0], &call, sizeof(rccl::rcclApiCall));
```

Reply:
Alternative to make it a little easier to comprehend.
```suggestion
    // compare only the fields after the pid field
    int result = memcmp(&(calls[0].pid)+1, &(call.pid)+1, sizeof(rccl::rcclApiCall)-sizeof(call.pid));
```